api/api.proto

```diff
@@ -1856,7 +1865,33 @@ service OpenStorageFilesystemCheck {
         body: "*"
       };
     }
-  }
+
+  // List all fsck created snapshots on volume
+  rpc ListSnapshots(SdkFilesystemCheckListSnapshotsRequest)
+    returns (SdkFilesystemCheckListSnapshotsResponse){
+      option(google.api.http) = {
+        get: "/v1/filesystem-check/list-snapshots"
+      };
+    }
+
+  // Delete all fsck created snapshots on volume
+  rpc DeleteSnapshots(SdkFilesystemCheckDeleteSnapshotsRequest)
+    returns (SdkFilesystemCheckDeleteSnapshotsResponse){
+      option(google.api.http) = {
+        post: "/v1/filesystem-check/delete-snapshots"
+        body: "*"
+      };
+    }
+
+  // List of all volumes which require fsck check/fix to be run
+  rpc ListVolumes(SdkFilesystemCheckListVolumesRequest)
+    returns (SdkFilesystemCheckListVolumesResponse){
+      option(google.api.http) = {
+        post: "/v1/filesystem-check/list-volumes"
```

Comment:
why is list-volumes a post request and not a get?

Reply:
Fixed